src/Razor/src/Microsoft.VisualStudio.RazorExtension/SyntaxVisualizer/SyntaxVisualizerControl.xaml.cs

```diff
@@ -121,6 +130,34 @@ public void ShowFormattingDocument()
 
         if (_fileUriProvider.TryGet(_activeWpfTextView.TextBuffer, out var hostDocumentUri))
         {
+            if (_languageServerFeatureOptions.UseRazorCohostServer)
+            {
+                var request = new DocumentContentsRequest
+                {
+                    TextDocument = new TextDocumentIdentifier { DocumentUri = hostDocumentUri },
+                    Kind = GeneratedDocumentKind.Formatting
+                };
+
+                var response = _joinableTaskFactory.Run(async () =>
+                {
+                    var lspResponse = await _lspRequestInvoker.ReinvokeRequestOnServerAsync<DocumentContentsRequest, DocumentContentsResponse>(
+                        _activeWpfTextView.TextBuffer,
+                        "razor/generatedDocumentContents",
+                        RazorLSPConstants.RoslynLanguageServerName,
+                        request,
+                        CancellationToken.None);
+
+                    return lspResponse?.Response;
+                });
+
+                if (response != null)
+                {
+                    OpenGeneratedCode(response.FilePath, response.Contents);
+                    return;
+                }
```

Comment:
Extract all of this to a separate method, called ShowGeneratedCode_Cohost, and have that method take 3 parameters:
1. ITextBuffer textBuffer (pass in _activeWpfTextView.TextBuffer in the call)
2. Uri hostDocumentUri
3. GeneratedDocumentKind kind

After calling the method, just return from this method